### PR TITLE
Added containerStyle props for platform specific searchbars

### PR DESCRIPTION
**Pull request by XeeshanAnsari**
Fixed issues for (#976)

### CODE REVIEW

File: src/searchbar/SearchBar-android.js [hunk 1]
```diff
@@ -61,10 +61,12 @@ class SearchBar extends Component {
   render() {
     const {
       clearIcon,
+      containerBgColor,
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-android.js]: There is already `containerStyle`

File: src/searchbar/SearchBar-android.js [hunk 2]
```diff
@@ -82,14 +84,23 @@ class SearchBar extends Component {
       />
     );
     return (
-      <View style={styles.container}>
+      <View
+        style={[
+          styles.container,
+          { backgroundColor: containerBgColor || '#FFF' },
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-android.js]: Apply the `containerStyle`

File: src/searchbar/SearchBar-android.js [hunk 3]
```diff
@@ -82,14 +84,23 @@ class SearchBar extends Component {
...
           {...attributes}
           onFocus={this.onFocus}
           onBlur={this.onBlur}
           onChangeText={this.onChangeText}
           ref={input => (this.input = input)}
-          inputStyle={[styles.input, inputStyle]}
+          inputStyle={[
+            styles.input,
+            inputBgColor && { backgroundColor: inputBgColor },
+            inputStyle,
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-android.js]: There is already an `inputStyle` for the background color

File: src/searchbar/SearchBar-android.js [hunk 4]
```diff
@@ -159,7 +172,6 @@ SearchBar.defaultProps = {
 
 const styles = StyleSheet.create({
   container: {
-    backgroundColor: 'white',
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-android.js]: Let the default color

File: src/searchbar/SearchBar-default.js [hunk 5]
```diff
@@ -50,7 +50,9 @@ class Search extends Component {
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-default.js]: The default SearchBar in `SearchBar-default.js` is already fully customizable. We just need to make the API more uniform between iOS/Android/Default searchbar, but I think you can let this file as it was. You are already able to change the background color.

File: src/searchbar/SearchBar-ios.js [hunk 6]
```diff
@@ -84,14 +86,23 @@ class SearchBar extends Component {
       <Ionicon size={20} name={'ios-search'} color={IOS_GRAY} />
     );
     return (
-      <View style={styles.container}>
+      <View
+        style={[
+          styles.container,
+          { backgroundColor: containerBgColor || '#f5f5f5' },
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-ios.js]: Same as in Android file, apply `containerStyle` here

File: src/searchbar/SearchBar-ios.js [hunk 7]
```diff
@@ -84,14 +86,23 @@ class SearchBar extends Component {
...
           {...attributes}
           onFocus={this.onFocus}
           onBlur={this.onBlur}
           onChangeText={this.onChangeText}
           ref={input => (this.input = input)}
-          inputStyle={[styles.input, inputStyle]}
+          inputStyle={[
+            styles.input,
+            { backgroundColor: inputBgColor || '#dcdce1' },
+            inputStyle,
```
Comment (xavier-villelegier) [inline src/searchbar/SearchBar-ios.js]: Same as in Android file, bg color is already customisable

**Comment by XeeshanAnsari** [comment]
Hi Thanks for quick response.
I have got your points so what will i have to do after fix that i want to confirm that should i new pull request ?

**Comment by xavier-villelegier** [comment]
@XeeshanAnsari Yea you can either open a new PR or fix this one, as you wish !

**Comment by XeeshanAnsari** [comment]
I have done so i'm going to new PR

File: src/searchbar/__tests__/__snapshots__/SearchBar.js.snap [hunk 8]
```diff
@@ -9,7 +9,7 @@ exports[`SearchBar wrapper component should render a default SearchBar if wrong
   onChangeText={[Function]}
   onClear={[Function]}
   placeholderTextColor="#86939e"
-  platform="wrong-platform"
```
Comment (xavier-villelegier) [inline src/searchbar/__tests__/__snapshots__/SearchBar.js.snap]: The goal of this test is to check that the default searchbar is rendered when a, unknown platform is specified (fallback pattern). Could you let `wrong-platform` or whatever different than `ios`, `android` or `default` ?

Comment (XeeshanAnsari) [reply src/searchbar/__tests__/__snapshots__/SearchBar.js.snap]: so should i change wrong-platform inside of default ?

Comment (xavier-villelegier) [reply src/searchbar/__tests__/__snapshots__/SearchBar.js.snap]: Yes, anything else that `ios`, `android` or `default`. `wrong-platform` was good.

Comment (XeeshanAnsari) [reply src/searchbar/__tests__/__snapshots__/SearchBar.js.snap]: okay dear

**Comment by AleksandrChernyavenko** [comment]
Any progress on this PR ?

File: src/searchbar/__tests__/SearchBar-ios.js [hunk 9]
```diff
@@ -12,81 +12,52 @@ describe('iOS SearchBar component', () => {
     expect(toJson(component)).toMatchSnapshot();
   });
 
-  it('should render with loading', () => {
```
Comment (xavier-villelegier) [inline src/searchbar/__tests__/SearchBar-ios.js]: Just add back this test and we're good to go

File: src/searchbar/__tests__/__snapshots__/SearchBar-ios.js.snap [hunk 10]
```diff
@@ -248,118 +257,19 @@ exports[`iOS SearchBar component should render with a custom search icon 1`] = `
 </View>
 `;
 
-exports[`iOS SearchBar component should render with loading 1`] = `
```
Comment (xavier-villelegier) [inline src/searchbar/__tests__/__snapshots__/SearchBar-ios.js.snap]: Same goes for this snapshot

**Comment by XeeshanAnsari** [comment]
I have added test.

**Comment by xavier-villelegier** [comment]
@XeeshanAnsari Thanks for your patience on this, it will be released in `v1.0.0-beta4`

**Pull request merged**